frontend/src/js/models/order.js

```diff
@@ -11,11 +11,15 @@ var Order = Backbone.Model.extend({
     email: ''
   },
 
-  preparePayment: function() {
+  preparePayment: function(onSuccess, onError) {
     // TODO: untested, I was offline when writing this, and connection is needed for paytrail
     $.post(this.urlRoot + '/' + this.get('id') + '/preparePayment',
       function(res) {
-        window.location.href = res.url;
+        if (res.err) {
```

Comment:
I'd move this to Store.jsx, it would probably make even onProceedToPayment easier to read, as it would have the if-clause instead of two anon functions. I'd actually remove this whole function and move also the ajax call to onProceedToPayment, as I don't really like these models so much anymore.